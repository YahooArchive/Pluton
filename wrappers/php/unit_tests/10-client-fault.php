#!/usr/local/bin/php
<?php

$client = new PlutonClient('fault');
$client->initialize("/does/not/exist");

if ($client->hasFault()) {
  print "client has fault after bogus initialize - good\n";
}
else {
  print "client does NOT have fault after bogus initialize - bad\n";
  exit(1);
}

$ftext = $client->getFault();
$fc = $client->getFaultCode();
$flong = $client->getFaultMessage("LONG", 1);
$fshort = $client->getFaultMessage("SHORT", 0);
print "Fault code=$fc Text=$ftext\n";
print "Short=$fshort\n";
print "Long=$flong\n";

if (strlen($flong) == 0) {
  print "Long message should not be zero length\n";
  exit(3);
}

if (strlen($fshort) == strlen($flong)) {
  print "Hmm expected short message to be shorter than long message\n";
  exit(4);
}

$client->initialize();
$client->reset();
if ($client->hasFault()) {
  print "client has fault after reset - bad\n";
  exit(5);
}

$request = new PlutonClientRequest();
$request->setRequestData("Some data");

$client->addRequest('system.echo.0.raw', $request);
if ($request->inProgress()) {
  print "Request is in progress - good\n";
}
else {
  print "Request is NOT in progress - bad\n";
  exit(6);
}

$client->reset();

if ($request->inProgress()) {
  print "Request is in progress - bad\n";
  exit(7);
}
else {
  print "Request is NOT in progress - good\n";
}

print "\nClient fault test ok\n";
?>
