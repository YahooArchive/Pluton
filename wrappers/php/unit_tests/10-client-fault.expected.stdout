client has fault after bogus initialize - good
Fault code=-33 Text=pluton::Fault=-33/-1 (openForMmapFailed:open() prior to mmap() of lookup Path failed) errno=2:No such file or directory '/does/not/exist' at pluton::client::initialize->initialize:325
Short=SHORT pluton::Fault=-33/-1 errno=2 '/does/not/exist'
Long=LONG pluton::Fault=-33/-1 (openForMmapFailed:open() prior to mmap() of lookup Path failed) errno=2:No such file or directory '/does/not/exist' at pluton::client::initialize->initialize:325
Request is in progress - good
Request is NOT in progress - good

Client fault test ok
